packages/protvista-zoomable/src/protvista-zoomable.js

```diff
@@ -246,12 +254,74 @@ class ProtvistaZoomable extends HTMLElement {
     this._ro = new ResizeObserver(this._onResize);
     this._ro.observe(this);
   }
+
+  _resetEventHandler(e) {
+    if (!e.target.closest(".feature")) {
+      this.dispatchEvent(this.createEvent("reset", null, true));
+    }
+  }
+
   getXFromSeqPosition(position) {
     return this.margin.left + this.xScale(position);
   }
   getSingleBaseWidth() {
     return this.xScale(2) - this.xScale(1);
   }
+
+  _getClickCoords() {
+    if (!d3Event) {
+      return null;
+    }
+    // const boundingRect = this.querySelector("svg").getBoundingClientRect();
+    // Note: it would be nice to also return the position of the bottom left of the feature
+    return [d3Event.pageX, d3Event.pageY];
+  }
+
+  createEvent(type, feature = null, withHighlight = false, start, end) {
+    const detail = {
+      eventtype: type,
+      coords: this._getClickCoords(),
+      feature: feature
+    };
+    if (withHighlight) {
+      detail.highlight = start && end ? `${start}:${end}` : null;
+    }
+    return new CustomEvent("change", {
+      detail: detail,
+      bubbles: true,
+      cancelable: true
+    });
+  }
+
+  bindEvents(d, el) {
```

Comment:
I'm changing the parameter names for `feature` and `element`just for readability.
```
  bindEvents(feature, element) {
```
I'll do the commit myself

Reply:
Sounds good. I’m sure there’s a way to bind `this` to the function instead of passing the element but d3 must be doing some context trickery as I could get it to work